File: cases/taint/csharp/closure-capture-negative/ClosureCaptureNegative.cs
using System;

namespace DataFlowBench;

static class ClosureCaptureNegative
{
    static string dfb_source() // DFB-SOURCE: closure-capture-input
    {
        return "tainted";
    }

    static void dfb_sink(string value) { } // DFB-SINK: closure-capture-sink

    static Action MakeHandler()
    {
        string tainted = dfb_source(); // DFB-WITNESS: closure-capture-store
        string captured = "clean";
        return () =>
        {
            dfb_sink(captured);
        };
    }

    static void Run()
    {
        Action handler = MakeHandler();
        handler();
    }
}

File: cases/taint/rust/closure-capture-negative/closure_capture_negative.rs
fn dfb_source() -> i32 { // DFB-SOURCE: closure-capture-input
    1
}

fn dfb_sink(value: i32) {} // DFB-SINK: closure-capture-sink

fn make_handler() -> Box<dyn Fn()> {
    let tainted = dfb_source(); // DFB-WITNESS: closure-capture-store
    let captured = 0;
    Box::new(move || {
        dfb_sink(captured);
    })
}

fn run() {
    let handler = make_handler();
    handler();
}
